### ISSUE TITLE
iFrame needs title attribute

### ISSUE THREAD

**Issue by weboverhauls** [ampproject/amp-by-example#679]
Framed content (iframe & frame elements) require a title attribute to describe the content. "This provides a label for the frame so users [of assistive technology] can determine which frame to enter and explore in detail."

WCAG 2.4.1 Bypass Blocks: A mechanism is available to bypass blocks of content that are repeated on multiple Web pages. (Level A)

There are 5 instances on the amp-iframe page
https://ampbyexample.com/components/amp-iframe/

Code example:
`<iframe class="i-amphtml-fill-content -amp-fill-content" name="amp_iframe0" allowfullscreen="" frameborder="0" sandbox="allow-scripts allow-same-origin allow-popups" src="https://player.vimeo.com/video/140261016#amp=1"></iframe>`

Suggested code:
`<iframe title="Netflix Branding: The Stack" class="i-amphtml-fill-content -amp-fill-content" name="amp_iframe0" allowfullscreen="" frameborder="0" sandbox="allow-scripts allow-same-origin allow-popups" src="https://player.vimeo.com/video/140261016#amp=1"></iframe>`

References:
https://www.w3.org/TR/2016/NOTE-UNDERSTANDING-WCAG20-20161007/navigation-mechanisms-skip.html
https://www.w3.org/TR/WCAG20-TECHS/H64.html

**Comment by lisaychuang** [comment]
Question - looking at the [amp-iframe page](https://ampbyexample.com/components/amp-iframe/) and [Github repo code](https://github.com/ampproject/amp-by-example/blob/master/src/20_Components/amp-iframe.html),  is this issue regarding adding a title attribute to the examples?

For example:

```
# current code
<amp-iframe width="500" height="281"
                          layout="responsive"
                          sandbox="allow-scripts allow-same-origin allow-popups"
                          allowfullscreen
                          frameborder="0"
                          src="https://player.vimeo.com/video/140261016">
  </amp-iframe>
```

```
# Suggested change
<amp-iframe width="500" height="281"
                          title="Netflix Branding: The Stack"
                          layout="responsive"
                          sandbox="allow-scripts allow-same-origin allow-popups"
                          allowfullscreen
                          frameborder="0"
                          src="https://player.vimeo.com/video/140261016">
  </amp-iframe>
```

I do not see the `title` attribute listed in the [amp-iframe docs](https://www.ampproject.org/docs/reference/components/amp-iframe#common-attributes) or the  [common attributes reference list](https://www.ampproject.org/docs/reference/common_attributes).

Can @aghassemi or @kul3r4 confirm if the `title` attribute is available?  I can update all examples once confirmed!

**Comment by sebastianbenz** [comment]
The title attribute is supported (tip: the [playground](https://ampbyexample.com/playground/#url=https%3A%2F%2Fampbyexample.com%2Fcomponents%2Famp-iframe%2Fsource%2F) is a quick way to check this). We should update the samples (there are probably other amp-iframe uses in the code base as well). Lisa, if you want to take a look that'd be fantastic!

**Comment by weboverhauls** [comment]
Yes, please add `title` to examples as well as documentation. Thank you.

**Comment by lisaychuang** [comment]
Updated samples on Amp By Example page and submitted a PR. 

To update reference documentation, it looks like I should be updating docs in [amphtml](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md) instead of [ampproject](https://github.com/ampproject/docs) docs directly?

> The docs in the reference section of the AMP Project site are not part of this repository; instead they are read from the [amphtml GitHub repository](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md) "source of truth."

**Comment by sebastianbenz** [comment]
Thanks for the PR. Regarding the reference docs, you're right, these need to be updated in the amphtml repository ([this](https://github.com/ampproject/amphtml/blob/master/extensions/amp-iframe/amp-iframe.md) is the file for the amp-iframe docs).

**Comment by lisaychuang** [comment]
Went through codebase and updated all repos with `<amp-iframe>` examples, see PR listed above.

Commented on outstanding issues #440 and #656 (which is merged, but this repo [Hotel](https://github.com/ampproject/amp-by-example/blob/42af747d668c28d66e1859ca7ef2daf48ee855c6/src/50_Samples_%2526_Templates/Hotel.html) is not added to AMP By Example -[ Samples & Templates page](https://ampbyexample.com/#samples_templates) yet.)

**Comment by sebastianbenz** [comment]
@lisaychuang  Awesome! Thanks a lot for going through the efforts of fixing all these!

**Comment by weboverhauls** [comment]
I see the title attribute in the documentation and the amp-iframe element, thank you. But in the final rendered markup, the title attribute remains on the on amp-iframe element, not the actual iframe.

The title attribute must be on the child iframe element (which has the video) not on the amp-iframe. I tested in Voiceover and confirmed that it's not working correctly.

**Comment by lesleslie** [comment]
I'm experiencing the same behavior as @weboverhauls. Is there a way for the amp-iframe element to set the title on the actual iframe or is there some other workaround?

**Comment by lesleslie** [comment]
Lighthouse flags it as an accessibility issue.

**Comment by catherinekorres** [comment]
I thought the title would apply to the child iframe element as well, but I'm getting the same lighthouse flag as @lesleslie since it's applying only to the amp-iframe element. Is there an upcoming solution?
